### PR TITLE
Add support for Bitlocker

### LINKED ISSUE #1047: Ask for password when opening bit locked drive for the first time
**Describe the bug**
Ask for password when opening bit locked drive for the first time. Now the app crashes

**To Reproduce**
1. Go to 'Files UWP'
2. Click on a drive that is bit locked (first time and it is locked)
3. App crashes

**Expected behavior**
Password should be prompted

**Desktop (please complete the following information):**
 - OS Version: [e.g. Windows 10 2004]
 - App version: [v0.10.1]

### PR DESCRIPTION
**Pull request by gave92**
Fixes #1047
This PR adds support for bitlocker drives.
When the user clicks on a drive on the sidebar, this PR:
- checks is the drive is bitlocker-encrypted and locked
- displays a dialog to ask for password
- attempts to unlock the drive

Some testing on a pc different from mine is probably in order 😄

### CODE REVIEW

File: Files/Dialogs/BitlockerDialog.xaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    x:Class="Files.Dialogs.BitlockerDialog"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local2="using:Files.Helpers"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d"
+    CornerRadius="4"
+    DefaultButton="Primary"
+    Title="Enter unlock password"
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml:11]:
```suggestion
    Title="Enter password to unlock drive"
```

File: Files/Dialogs/BitlockerDialog.xaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    BorderThickness="0"
+    PrimaryButtonClick="NameDialog_PrimaryButtonClick"
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml:13]:
```suggestion
    PrimaryButtonClick="BitlockerDialog_PrimaryButtonClick"
```

File: Files/Dialogs/BitlockerDialog.xaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    x:Name="NameDialog"
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml:14]:
```suggestion
    x:Name="BitlockerDialog"
```

File: Files/Dialogs/BitlockerDialog.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        public string storedPasswordInput;
+
+        public BitlockerDialog(string drive)
+        {
+            this.InitializeComponent();
+            inputBox = PasswordInput;
+            IsPrimaryButtonEnabled = false;
+        }
+
+        private void NameDialog_PrimaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml.cs:20]:
```suggestion
        private void BitlockerDialog_PrimaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
```

File: Files/Dialogs/BitlockerDialog.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            {
+                IsPrimaryButtonEnabled = false;
+            }
+            else
+            {
+                IsPrimaryButtonEnabled = true;
+            }
+        }
+
+        private void NameDialog_KeyDown(object sender, Windows.UI.Xaml.Input.KeyRoutedEventArgs e)
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml.cs:39]:
```suggestion
        private void BitlockerDialog_KeyDown(object sender, Windows.UI.Xaml.Input.KeyRoutedEventArgs e)
```

File: Files/Strings/en-US/Resources.resw [hunk 6]
```diff
@@ -753,4 +753,16 @@
...
     <value>B</value>
   </data>
+  <data name="BitlockerDialog.PrimaryButtonText" xml:space="preserve">
+    <value>Unlock</value>
+  </data>
+  <data name="BitlockerDialog.SecondaryButtonText" xml:space="preserve">
+    <value>Cancel</value>
+  </data>
+  <data name="BitlockerDialog.Title" xml:space="preserve">
+    <value>Enter unlock password</value>
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:763]:
```suggestion
    <value>Enter password to unlock drive</value>
```

File: Files/Strings/en-US/Resources.resw [hunk 7]
```diff
@@ -753,4 +753,16 @@
...
+  </data>
+  <data name="BitlockerDialogInputText.PlaceholderText" xml:space="preserve">
+    <value>Enter drive unlock password</value>
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:766]:
```suggestion
    <value>Password</value>
```

File: Files/Dialogs/BitlockerDialog.xaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    KeyDown="NameDialog_KeyDown">
+
+    <Grid MinWidth="300">
+        <StackPanel
+            Orientation="Vertical"
+            Spacing="4">
+            <PasswordBox 
+                x:Name="PasswordInput"
+                x:Uid="BitlockerDialogInputText"
+                PlaceholderText="Enter drive unlock password"
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml:28]:
```suggestion
                PlaceholderText="Password"
```

**Review by yair100 — CHANGES_REQUESTED** [review]

**Comment by yaichenbaum** [comment]
@gave92 This is great work! I made a few suggestions in order to improve the PR, once those are resolved we will merge the pull request.

File: Files/Dialogs/BitlockerDialog.xaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    xmlns:local2="using:Files.Helpers"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d"
+    CornerRadius="4"
+    DefaultButton="Primary"
+    Title="Enter password to unlock the drive"
+    BorderThickness="0"
+    PrimaryButtonClick="BitlockerDialog_PrimaryButtonClick"
+    x:Name="BitlockerDriveDialog"
```
Comment (yaichenbaum) [inline Files/Dialogs/BitlockerDialog.xaml:14]:
```suggestion
    x:Name="BitlockerDialog"
```

**Review by yair100 — CHANGES_REQUESTED** [review]
The dialog should be declared in App.cs the same way the other dialogs are.

File: Files/Dialogs/BitlockerDialog.xaml (hunk 9 above)
Comment (gave92) [reply Files/Dialogs/BitlockerDialog.xaml:14]: Apparently the name can't be the same of the class (it won't compile).
Also the others differ:
RenameDialog => NameDialog
PropertiesDialog => PropertiesDialogMarkup

**Review by yair100 — APPROVED** [review]

File: Files/View Models/ItemViewModel.cs [hunk 10]
```diff
@@ -863,6 +880,16 @@ public async Task EnumerateItemsFromStandardFolder(string path)
 
         }
 
+        private async Task<bool> CheckBitlockerStatus(StorageFolder rootFolder)
+        {
+            if (Path.IsPathRooted(WorkingDirectory) && Path.GetPathRoot(WorkingDirectory) == WorkingDirectory)
+            {
+                IDictionary<string, object> extraProperties = await rootFolder.Properties.RetrievePropertiesAsync(new string[] { "System.Volume.BitLockerProtection" });
+                return (int?)extraProperties["System.Volume.BitLockerProtection"] == 6; // Drive is bitlocker protected and locked
```
Comment (tsvietOK) [inline Files/View Models/ItemViewModel.cs:888]: Are you sure about 6 means drive is protected? Che this [link](https://stackoverflow.com/a/41310139/12258600).

Comment (gave92) [reply Files/View Models/ItemViewModel.cs:888]: Yes that's the link that helped me implement this. I tested it on my pc on a bitlocker drive and the status is 6 when locked, 1 when unlocked and 2 when not protected. If someone else could test it too I would feel more confident.

**Review by yair100 — APPROVED** [review]

**Comment by tsvietOK** [comment]
Please, resolve conflicts.
When i typed wrong password, it showed me empty folder. Is it possible to check password?

**Comment by gave92** [comment]
I can ask for pw again if the drive does not unlock. Currently you have to click on the sidebar item again to get prompted for pw again.

**Review by yair100 — APPROVED** [review]

**Comment by gave92** [comment]
@yair100 please wait before merging. There's a bug I need to fix :)

**Comment by gave92** [comment]
@yair100 ok should be good now.
Changes:
- Fix crash of fulltrust process when UAC is denied by user
- Add "shield"/"admin" icon to bitlocker unlock dialog -------> is this ok?
- Add dialog when wrong password is entered

File: Files/Strings/en-US/Resources.resw [hunk 11]
```diff
@@ -756,4 +756,22 @@
...
+    <value>Cancel</value>
+  </data>
+  <data name="BitlockerDialog.Title" xml:space="preserve">
+    <value>Enter password to unlock the drive</value>
+  </data>
+  <data name="BitlockerDialogInputText.PlaceholderText" xml:space="preserve">
+    <value>Password</value>
+  </data>
+  <data name="BitlockerInvalidPwDialog.Text" xml:space="preserve">
+    <value>Could not unlock the drive. Wrong password?</value>
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:772]:
```suggestion
    <value>Please check the password and try again</value>
```

File: Files/Interacts/Interaction.cs [hunk 12]
```diff
@@ -656,7 +656,7 @@ public async void DeleteItem_Click(object sender, RoutedEventArgs e)
             }
             catch (IOException)
             {
-                if (await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("FileInUseDeleteDialog.Title"), ResourceController.GetTranslation("FileInUseDeleteDialog.Text"), ResourceController.GetTranslation("FileInUseDeleteDialog.PrimaryButtonText"), ResourceController.GetTranslation("FileInUseDeleteDialog.SecondaryButtonText")))
+                if (await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("FileInUseDeleteDialog/Title"), ResourceController.GetTranslation("FileInUseDeleteDialog/Text"), ResourceController.GetTranslation("FileInUseDeleteDialog/PrimaryButtonText"), ResourceController.GetTranslation("FileInUseDeleteDialog/SecondaryButtonText")))
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:659]: You are doing the same changes as in your another [PR](https://github.com/files-community/files-uwp/pull/1137/files#diff-0d4ba4bdc6e2e429e16d3f061cb662a9). You have to choose which one you want to leave.

File: Files/MultilingualResources/Files.de-DE.xlf [hunk 13]
```diff
@@ -865,6 +865,30 @@
...
+        <trans-unit id="BitlockerDialogInputText.PlaceholderText" translate="yes" xml:space="preserve">
+          <source>Password</source>
+          <target state="new">Password</target>
+        </trans-unit>
+        <trans-unit id="BitlockerInvalidPwDialog.Text" translate="yes" xml:space="preserve">
+          <source>Please check the password and try again.</source>
+          <target state="new">Please check the password and try again.</target>
+        </trans-unit>
+        <trans-unit id="BitlockerInvalidPwDialog.Title" translate="yes" xml:space="preserve">
+          <source>Bitlocker error</source>
```
Comment (tsvietOK) [inline Files/MultilingualResources/Files.de-DE.xlf:889]:
```suggestion
          <source>BitLocker error</source>
```
Change all occurrences of this word.

**Review by tsvietOK — CHANGES_REQUESTED** [review]
And check for duplicated changes between your PR's.

**Review by yair100 — APPROVED** [review]

**Review by tsvietOK — APPROVED** [review]

**Pull request merged**